app/controllers/admin/collections_controller.rb

```diff
@@ -298,12 +298,10 @@ def collection_params
     params.permit(:admin_collection => [:name, :description, :unit, :managers => []])[:admin_collection]
   end
 
-  def resize_uploaded_poster(file)
-    image = ::MiniMagick::Image.open(file)
-    image.thumbnail '700x700>'
-    image.format 'png'
-    image.tempfile.open.read
-  rescue MiniMagick::Invalid
-    return nil
+  def check_image_compliance(poster_path)
+    type = FastImage.type(poster_path) == :png
+    # Size derived from width and aspect ratio from JS code, assets/javascript/crop_upload.js:60-63
+    size = FastImage.size(poster_path) == [700, 560] # [width, height]
+    type && size
```

Comment:
FastImage.type() and FastImage.size() both open and read the file so I think it would be better to instantiate a FastImage object to avoid the second open and read.
```suggestion
    fastimage = FastImage.new(poster_path)
    # Size derived from width and aspect ratio from JS code, assets/javascript/crop_upload.js:60-63
    fastimage.type == :png && fastimage.size == [700, 560] # [width, height]
```